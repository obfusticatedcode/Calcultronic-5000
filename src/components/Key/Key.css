.Key__container {
  align-items: center;
  display: flex;
  height: 25%;
  justify-content: center;
  transition: background-color 0.3s linear;
}

.Key__container:hover {
  cursor: pointer;
}

.Key__operatorKey {
  background-color: var(--action-key-color);
  width: 100%;
}

.Key__operatorKey:hover {
  background-color: var(--action-key-color-hover);
}

.Key__numberKey {
  background-color: var(--number-key-color);
  width: calc(100% / 3);
}

.Key__numberKey:hover {
  background-color: var(--number-key-color-hover);
}

.Key__submitKey {
  background-color: var(--submit-key-color);
  height: 100%;
  width: 100%;
}

.Key__submitKey:hover {
  background-color: var(--submit-key-color-hover);
}

.Key__value {
  color: var(--key-text-color);
  font-family: var(--main-font);
  font-size: var(--key-text-size);
  font-weight: var(--key-text-weight);
}
