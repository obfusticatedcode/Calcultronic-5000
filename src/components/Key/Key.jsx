import React from "react";
import PropTypes from "prop-types";
import "./Key.css";

const Key = props => {
  const { keyAction, keyType, keyValue } = props;
  const handleClick = () => keyAction(keyValue);

  return (
    <div className={`Key__container ${keyType}`} onClick={handleClick}>
      <p className="Key__value">{keyValue}</p>
    </div>
  );
};

Key.propsTypes = {
  keyAction: PropTypes.func.isRequired,
  keyType: PropTypes.string.isRequired,
  keyValue: PropTypes.string.isRequired
};
export default Key;
