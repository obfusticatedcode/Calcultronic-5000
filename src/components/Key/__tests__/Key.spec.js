import React from "react";
import { shallow } from "enzyme";
import Key from "../Key";

const keyAction = jest.fn();
const defaultProps = {
  keyAction,
  keyType: "",
  keyValue: ""
};
const setup = (props = {}) => shallow(<Key {...defaultProps} {...props} />);

describe("<Key/>", () => {
  let wrapper = null;
  beforeEach(() => (wrapper = setup({})));
  it("should render a <div />", () => {
    expect(wrapper.find("div").length).toEqual(1);
  });

  it("should render the value of keyValue", () => {
    wrapper.setProps({ keyValue: "demo" });
    expect(wrapper.text()).toEqual("demo");
  });

  it("should render correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });
});
