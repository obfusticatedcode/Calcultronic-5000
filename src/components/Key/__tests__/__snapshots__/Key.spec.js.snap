// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Key/> should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Key
    keyAction={[MockFunction]}
    keyType=""
    keyValue=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <p
        className="Key__value"
      >
        
      </p>,
      "className": "Key__container ",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "",
        "className": "Key__value",
      },
      "ref": null,
      "rendered": "",
      "type": "p",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <p
          className="Key__value"
        >
          
        </p>,
        "className": "Key__container ",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "",
          "className": "Key__value",
        },
        "ref": null,
        "rendered": "",
        "type": "p",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
